tests/Type/data/auth.php

```diff
@@ -20,4 +20,5 @@ function test(User $user): void
     assertType('App\Admin|null', Auth::guard('admin')->user());
     assertType(SessionGuard::class, Auth::guard('session'));
     assertType('null', Auth::guard()->login($user));
+    assertType('App\Admin|App\User', Auth::authenticate());
```

Comment:
Can you add tests for the guards?

```suggestion
    assertType('App\Admin|App\User', Auth::authenticate());
    assertType('App\User', Auth::guard('web')->authenticate());
    assertType('App\Admin', Auth::guard('admin')->authenticate());
```

Reply:
I'll revisit this one as `Auth::guard('web')` has the type `Illuminate\Contracts\Auth\StatefulGuard` right now which doesn't have the method in the interface so phpstan doesn't provide any way to bind a ReturnType extension. 
If I improve the return type of `Auth::guard('web')` to the concrete implementation then I should be able to attach a return type to `Auth::guard('web')->authenticate()`

Reply:
Ah that makes sense. Laravel has a really bad habit of typing interfaces and then proceeding to define and use extra methods on the runtime implementation that don't adhere to the contract---very frustrating from a static analysis point of view.

Reply:
All sorted, there was a bug in the return type provider of `Auth::guard($x)`, it was using `$x` as the driver, not what was set in the config.

Tweaked a couple of bits further so the same applies to `auth($x)` as well.